frontEnd/tickets-app/src/components/MovieSession/MovieSession.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from "react";
+import LoadingButton from "@mui/lab/LoadingButton";
+import { Grid } from "@mui/material";
+import "./MovieSession.scss";
+import { filmsApi } from "../../api/filmsApi";
+
+const MovieSession = ({ cinema, halls, cbSetShowSession }) => {
+  const [hallsEl, setHallsEl] = useState(false);
+  const [loading, setLoading] = useState(false);
+  useEffect(() => {
+    if (halls !== false) {
+      let arrHallsEl = [];
+      let firsttime = true;
+      for (let hall in halls) {
+        let arrButSessions = halls[hall].map((e) => (
+          <LoadingButton
+            variant="contained"
+            key={e._id + "" + e.session._id}
+            className="sessions"
+            onClick={async () => {
```

Comment:
move it to a separated function, it's unreadable to define functions in props directly